tests/PhpStan/FunctionStrictModeRuleTest.php

```diff
@@ -31,11 +31,13 @@ public function shouldSuccessProcess(): void
         $this->analyse(
             [__DIR__.'/Resources/function-strict-mode.php'],
             [
-                ['The function in_array must be used in strict mode.', 16],
-                ['The function in_array must be used in strict mode.', 17],
-                ['The function in_array must be used in strict mode.', 18],
-                ['The function in_array must be used in strict mode.', 21],
-                ['The function in_array must be used in strict mode.', 23],
+                ['The function in_array must be used in strict mode.', 26],
+                ['The function in_array must be used in strict mode.', 27],
+                ['The function in_array must be used in strict mode.', 28],
+                ['The function in_array must be used in strict mode.', 31],
+                ['The function in_array must be used in strict mode.', 33],
+                ['The function in_array must be used in strict mode.', 35],
```

Comment:
No correct, because if we check object, object also checks hash too, as result, we can not use strict mode for enums, because enums is objects. We must use strict checks for scalar, as example `1 !== "1"` or `true !== 1`.

```php
// correct
\in_array(1, [1, 2, 3], true); // Because check scalars, and 1 !== "1".

$a = "1";
\in_array($a, [1, 2, 3], true);

$enum = SomeEnum::Bla;
\in_array($enum, [SomeEnum::Bla, SomeEnum::Foo, SomeEnum::Bar]); // correct because enum is not scalar and php correct check by "hashes".
```